ProjFS.Mac/PrjFSKextTests/MockProc.cpp

```diff
@@ -25,32 +34,81 @@ proc_t vfs_context_proc(vfs_context_t ctx)
 
 proc_t proc_self(void)
 {
-    return nullptr;
+    // Since this function takes no parameters, it will return the first entry in the map, if one exists
```

Comment:
Since it's an ordered map, it's not arbitrary.  The user would have added it in the test order.  

We do need the ability to change this value for testing.

I don't love the SetSelfPid option either honestly.  We could do it, if we think it adds clarity.

Reply:
Updated so eventhing is done thru 'add's and we lookup in a map